### DISCUSSION TITLE
"Open in AR" with QR-code?

### DISCUSSION THREAD

**Discussion by gerritkk** [google/model-viewer#2718] [Q&A]
Is it possible to instead of a button have a QR-code on pc which after being scanned with a phone, basically does the same as the button that appears when ur on a mobile device?

**Comment by elalish** [comment]
Yes, try our "Deploy Mobile" feature on our editor: https://modelviewer.dev/editor/

This is tricky because it syncs state between the devices. If you just want to view the model, then just point the QR code to the page's own URL and if  <model-viewer> is there with the `ar` attribute, it'll just work.

**Reply by WadeMorrison** [reply]
@elalish 
I just looked into this as this is something I'm really trying to implement into my configurator. Is this possible to use for a configured product? Such as the client chose a size, and color (3 sizes to choose from, and 10+ colors to choose from), and then hit the QR button, scanned it, and then launch it automatically in AR with those defined selections?

So far I only have a QR code generator to grab the document's URL on desktop, which would then just lead the client to the configurator on mobile and then have to re-choose all the selections again. I've only got like a month or two worth of HTML/CSS/JS scripting so it's all relatively new to me and not sure what keywords I need to search.

Hopefully this makes sense? I can provide examples of what I'm looking for/what I have currently?

**Reply by elalish** [reply]
You might be able to simplify this flow quite a bit by skipping the QR code entirely. Look at these examples, which allow configuration directly on a mobile device, even during an AR session: https://modelviewer.dev/examples/scenegraph/

**Reply by WadeMorrison** [reply]
Oh yeah, I looked through that as well, however, a majority of our customer's are on IOS, and from what I saw IOS doesn't support that correct? I'm basically after what #1820 did with their customizer. 

Here is WIP of what I have going on currently: [https://mallorychest.s3.us-east-2.amazonaws.com/index.html](https://mallorychest.s3.us-east-2.amazonaws.com/index.html) (Sizes are wrong, and other bugs as well, just trying to get a good template made for the rest of our products)

Thank you for replying back!

**Reply by elalish** [reply]
In fact that will also work without modification if you use our [USDZ auto-generation](https://modelviewer.dev/docs/#entrydocs-augmentedreality-attributes-arModes) feature. Since we  export the scene right when you launch Quick Look, it means any variant or scene-graph modifications you have made will be taken into account. Try [these](https://modelviewer.dev/examples/scenegraph/#swapTextures) examples on  your iPhone. You just  need  to be a little careful with your glTF since not all features are supported in USDZ.

**Reply by WadeMorrison** [reply]
Sorry, I don't think I'm explaining what I am after very well haha.. I got the auto-generation tool already with the configurator, I was just wondering about the AR button. Since the AR slot icon is hidden on desktop, I wanted to create a QR code that the customer could scan (if configured from desktop) that would then launch the configured model into AR. I read somewhere about explicit intent, but IOS doesn't support that from what I had read?

**Reply by elalish** [reply]
I guess my point is, why would you start on desktop when you could configure on mobile directly? If you want to though, I think the easiest way is just to encode your state as query params in the URL.

**Reply by WadeMorrison** [reply]
Mainly in case our customer's shopped on desktop versus mobile first. I know the majority of them shop on mobile, I just wanted to try and have this implemented on the off chance that the customer configured on desktop first is all. 

But okay, I will do some more research on how to encode the query parameters in the URL. I appreciate the help and responsiveness!
